test/com/amazon/ion/TestUtils.java

```diff
@@ -567,8 +571,18 @@ public static boolean symbolTableEquals(final SymbolTable first, final SymbolTab
     public static class BinaryIonAppender {
         private final ByteArrayOutputStream out = new ByteArrayOutputStream();
 
+        public BinaryIonAppender(int minorVersion) throws Exception {
```

Comment:
Could we just have an enum of the different Ion versions so that it's not possible to provide an invalid version, or are there performance implications to doing that?

Reply:
There are no performance implications in this test code, but it's just us writing the tests, so I don't think it's super important for us to protect ourselves from providing an invalid version to this utility.

If the question is intended to apply broadly to the source, there are some performance implications when we get into, say, branching vs. dynamic dispatch. For example, if we wanted a non-trivial enum that contained abstract methods to be overridden by each version. This is similar to how I wrote the first Ion 1.1 prototype, and I found simple branching to be much more efficient. We could have a simple enum that just enumerates the acceptable versions, but on the read side we still have to validate the version that was declared to the encoding, then map to the enum. After validation, I'm not sure if it really matters whether we compare an int or an enum value on the branches. Maybe an enum starts being more clean once we have another major version, but we're a ways off from that.

Reply:
An enum seems like a good idea to me.